File: src/backend/Repository/EFRepo/Entities/Subscription.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace EFRepo.Entities
{
    /// <summary>
    /// Subscription
    /// </summary>
    public class Subscription
    {
        /// <summary>
        /// Identifier
        /// </summary>
        public int SubscriptionId { get; set; }

        public int UserId { get; set; }

        /// <summary>
        /// User identifier
        /// </summary>
        [ForeignKey(nameof(UserId))]
        public User User { get; set; }

        public int TaskId { get; set; }

        /// <summary>
        /// Task identifier
        /// </summary>
        [ForeignKey(nameof(TaskId))]
        public Task Task { get; set; }

        /// <summary>
        /// Token
        /// </summary>
        public string Token { get; set; }
    }
}

File: src/backend/Notifier/src/main/java/org/aidmind/reminder/model/Subscription.java
package org.aidmind.reminder.model;

public class Subscription {
        private int subscriptionId;
        private int userId;
        private int taskId;

    public Subscription(int subscriptionId, int userId, int taskId) {
        this.subscriptionId = subscriptionId;
        this.userId = userId;
        this.taskId = taskId;
    }

    public int getSubscriptionId() {
        return subscriptionId;
    }

    public int getUserId() {
        return userId;
    }

    public int getTaskId() {
        return taskId;
    }
}
